pages/button-group/test.page.tsx

```diff
@@ -3,177 +3,239 @@
+    text: 'More actions',
+    loading: !!(loadingId && ['cut', 'paste', 'edit', 'open', 'search'].includes(loadingId)),
+    items: [
+      {
+        id: 'cut',
+        iconName: 'delete-marker',
+        text: 'Cut',
+      },
+      {
+        id: 'paste',
+        iconName: 'add-plus',
+        text: 'Paste',
+        disabled: true,
+        disabledReason: 'No content to paste',
+      },
+      {
+        text: 'Misc',
+        items: [
+          { id: 'edit', iconName: 'edit', text: 'Edit' },
+          { id: 'open', iconName: 'file-open', text: 'Open' },
+          { id: 'search', iconName: 'search', text: 'Search' },
+        ],
+      },
+    ],
+  };
+
+  function canRenderItem(item: ButtonGroupProps.ItemOrGroup) {
+    if (item.type === 'group' && item.text === 'Send' && !canSend) {
+      return false;
+    }
+    return true;
+  }
+  const items = [
     feedbackGroup,
+    favoriteGroup,
+    sendGroup,
     copyButton,
     addButton,
-    sendButton,
-    redoButton,
     removeButton,
+    redoButton,
     moreActionsMenu,
-  ]);
-
-  const onItemClick: ButtonGroupProps['onItemClick'] = event => {
-    document.querySelector('#last-clicked')!.textContent = event.detail.id;
+  ].filter(canRenderItem);
 
-    if (event.detail.id === 'dislike') {
-      setItems(prev => prev.filter(item => item.type !== 'group'));
+  const onItemClick: ButtonGroupProps['onItemClick'] = ({ detail }) => {
+    function addLog(text: string) {
+      const entry = document.createElement('div');
+      entry.textContent = text;
+      document.querySelector('#log')!.append(entry);
     }
-    if (event.detail.id === 'redo') {
-      setItems(prev =>
-        prev.map(item => (item.type === 'icon-button' && item.id === 'redo' ? { ...item, disabled: true } : item))
-      );
-    }
-    if (event.detail.id === 'remove') {
-      setItems(prev =>
-        prev.map(item => (item.type === 'icon-button' && item.id === 'remove' ? { ...item, loading: true } : item))
-      );
+
+    function syncAction(action?: () => void) {
+      addLog(detail.id);
+      action?.();
     }
-  };
 
-  const onFocusOnCopyButtonClick = () => {
-    ref.current?.focus('copy');
-  };
+    function asyncAction(action?: () => void) {
+      setLoading(detail.id);
+      const callback = () => {
+        setLoading(null);
+        addLog(detail.id);
+        action?.();
+      };
+      if (manualServerMock) {
+        window.__pendingCallbacks.push(callback);
+      } else {
+        setTimeout(callback, 1000);
+      }
+    }
 
-  const onFocusOnMoreActionsButtonClick = () => {
-    ref.current?.focus('more-actions');
+    switch (detail.id) {
+      case 'like':
+      case 'dislike':
+        return syncAction(() => setFeedback(prev => (prev !== detail.id ? (detail.id as 'like' | 'dislike') : 'none')));
+      case 'favorite':
+        return asyncAction(() => setFavorite(prev => !prev));
+      case 'send':
+        return syncAction(() => setCanSend(false));
```

Comment:
Do we want the button to disappear after clicking the send, or should we show another state of the same button?
| Before click | After click |
| --- | --- |
| <img width="524" alt="image" src="https://github.com/user-attachments/assets/308d90b0-0d93-4c24-a636-89671512e3c4"> | <img width="524" alt="image" src="https://github.com/user-attachments/assets/2e2430a5-599f-44e5-9abb-c89e5e8bf1bb"> |

Reply:
The demo features two different behaviours. For the send button we hide it and for the redo button we make it disabled. I think the disappearing button is a very corner case but I think it is still worths having for completeness.